src/Atc.Rest.Client/EndpointResult.cs

```diff
@@ -0,0 +1,55 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
+namespace Atc.Rest.Client
+{
+    public class EndpointResult<TResponseContent> : EndpointResponse
```

Comment:
If we include this type in the project, we have to make the `FailedContent` a generic parameter. Other users might not be using `ProblemDetails`, and there might be binary incompatibilities between the version included in the library, and the one that others might want to use.

```suggestion
    public class EndpointResult<TSuccessContent, TErrorContent> : EndpointResponse
```